mhlib/mhConfig.go

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2018 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mhlib
+
+import (
+	"github.com/imdario/mergo"
+)
+
+// MHConfigFile is the structure of a mh configuration file.
+type MHConfigFile struct {
+	MH         MHConfig   `yaml:"mh"`
+	Apps       Apps       `yaml:"apps"`
+	AppSources AppSources `yaml:"appSources"`
+}
+
+// MHConfig is a set of options used during app deployment.
+type MHConfig struct {
+	Maintainers   []string `yaml:"maintainers"`
+	Namespace     string   `yaml:"namespace"`
+	PrintRendered bool     `yaml:"printRendered"`
+	Simulate      bool     `yaml:"simulate"`
+	TargetContext string   `yaml:"targetContext"`
+	Team          string   `yaml:"team"`
+}
+
+// DefaultMHConfig is the default mh config and will most likely be modified
+// during execution with the following sources by rising priority:
+//
+// 1. This default configuration
+// 2. The "mh" key in the configuration file
+// 3. Environment variables starting with "MH_"
+// 4. Command line flags
+// 5. app-specific overrides in MH_CONFIG.
```

Comment:
I'm not entirely happy with this order, moving 5. up between 2. and 3. might be a better solution. TBD

Reply:
Your described order is what I would expect as well.